resolver/examples/global_resolver.rs

```diff
@@ -0,0 +1,141 @@
+#[macro_use]
+extern crate lazy_static;
+extern crate futures;
+extern crate tokio;
+extern crate tokio_io;
+extern crate trust_dns_resolver;
+
+use std::io;
+use std::net::SocketAddr;
+
+use futures::Future;
+use tokio_io::IoFuture;
+use trust_dns_resolver::{IntoName, ResolverFuture, TryParseIp};
+
+// This is an example of registering a static global resolver into any system.
+//
+// We may want to create a GlobalResolver as part of the Resolver library
+//   in the mean time, this example has the necessary steps to do so.
+//
+// Thank you to @zonyitoo for the original example.
+
+lazy_static! {
+    // First we need to setup the global Resolver
+    static ref GLOBAL_DNS_RESOLVER: ResolverFuture = {
+        use std::sync::{Arc, Mutex, Condvar};
+        use std::thread;
+
+        // We'll be using this condvar to get the Resolver from the thread...
+        let pair = Arc::new((Mutex::new(None::<ResolverFuture>), Condvar::new()));
+        let pair2 = pair.clone();
+
+
+        // Spawn the runtime to a new thread...
+        //
+        // This thread will manage the actual resolution runtime
+        thread::spawn(move || {
+            // A runtime for this new thread
+            let mut runtime = tokio::runtime::current_thread::Runtime::new().expect("failed to launch Runtime");
+
+            // our platform independent future, result, see next blocks
+            let future;
+
+            // To make this independent, if targeting macOS, BSD, Linux, or Windows, we can use the system's configuration:
+            #[cfg(any(unix, windows))]
+            {
+                // use the system resolver configuration
+                future = ResolverFuture::from_system_conf().expect("Failed to create ResolverFuture");
+            }
+
+            // For other operating systems, we can use one of the preconfigured definitions
+            #[cfg(not(any(unix, windows)))]
+            {
+                // Directly reference the config types
+                use trust_dns_resolver::config::{ResolverConfig, ResolverOpts};
+
+                // Get a new resolver with the google nameservers as the upstream recursive resolvers
+                future = ResolverFuture::new(ResolverConfig::google(), ResolverOpts::default()).expect("Failed to create ResolverFuture");
```

Comment:
BUG. `ResolverFuture::new` doesn't return a `Result`

Reply:
Gah. I forgot to add this to the appveyor build. Would have caught this.

Thanks.